ee/cis/macos-14/cis-policy-queries.yml

```diff
@@ -3448,6 +3448,39 @@ spec:
 ---
 apiVersion: v1
 kind: policy
+spec:
+  name: CIS - Ensure Show Status Bar Is Enabled (MDM Required)
+  platforms: macOS
+  platform: darwin
+  description: |
+    The Status Bar in Safari shows the full URL of any link on hover. It protects the user from visiting sites where the domain has been obfuscated by allowing the user to review whether the link points to an unexpected location.
+  resolution: |
+    Automated method:
+    Profile Method:
+    Create or edit a configuration profile with the following information:
+      1. The PayloadType string is com.apple.Safari
+      2. The key to include is ShowOverlayStatusBar 
+      3. The key must be set to: <true/>
+  query: |
+    SELECT 1 WHERE 
+      EXISTS (
+        SELECT 1 FROM managed_policies WHERE 
+            domain='com.apple.Safari' AND 
+            name='ShowOverlayStatusBar' AND 
+            (value = 1 OR value = 'true')
```

Comment:
@lucasmrod, when checking the list of users in `managed_policies` I find `username = NULL` for which the field `ShowOverlayStatusBar = 1` is not deployed by the profile.

Thus this will not work.
```
SELECT 1 WHERE NOT EXISTS (
      SELECT 1 FROM users AS u
      LEFT JOIN (
        SELECT username FROM managed_policies WHERE
            domain=‘com.apple.Safari’ AND
            name=‘ShowOverlayStatusBar’ AND
            (value != 1 OR value != ‘true’)
      ) AS p
      ON p.username = u.username OR p.username IS NULL
    );
```


The problem remains that the profile only deploys the desired outcome to one username.
The query as it is now (I removed `AND username=''`) does work here.
TMWYT

Reply:
I ran the following steps:
1. Installed https://github.com/fleetdm/fleet/blob/main/it-and-security/lib/configuration-profiles/macos-ensure-show-status-bar-is-enabled.mobileconfig on a macOS 14 VM with one user account "lucas".
2. `managed_policies` returns the applied setting as expected and with `username="lucas"`
3. I then created another user "gandalf".
4. `managed_policies` still returns the same results as (2).
5. I log out as "lucas" and log in as "gandalf".
6. `managed_policies` now returns two rows with `domain='com.apple.Safari', name='ShowOverlayStatusBar'` (one with `username='lucas'` and one with `username='gandalf'`).
![Screenshot 2024-10-17 at 7 04 41 AM](https://github.com/user-attachments/assets/6d7f6a24-f614-47b5-9cf2-83c54954172f)

This means the `domain='com.apple.Safari', name='ShowOverlayStatusBar'` setting is system wide and, once applied, macOS makes sure all users have such setting applied. Thus, knowing this, the query that we need is now actually simpler:
```sql
SELECT 1 FROM managed_policies WHERE domain='com.apple.Safari' AND name='ShowOverlayStatusBar' AND (value = 1 OR value = 'true');
```

Reply:
@lucasmrod, In that case the existing query is G2G.
I will merge soon.